src/integrator_interface.jl

```diff
@@ -81,44 +81,42 @@ function perform_step!(integrator::DDEIntegrator)
                             integrator.fixedpoint_reltol)
             end
 
-            # stop fixed-point iteration when residuals are small or maximal number of steps is exceeded
+            # stop fixed-point iteration when residuals are small or maximal number of steps
+            # is exceeded
             fixedpointEEst = integrator.fixedpoint_norm(integrator.resid)
             if fixedpointEEst < 1 || numiters > integrator.max_fixedpoint_iters
                 break
             end
 
-            # special updates of ODE integrator after the first iteration step
+            # special update of ODE integrator after the first iteration step
             # to use interpolation of ODE integrator in the next iterations
             # when evaluating the history function
             if numiters == 1
-                integrator.integrator.tprev = integrator.t
                 integrator.integrator.t = integrator.t + integrator.dt
-                integrator.integrator.uprev = integrator.u
             end
 
-            # update ODE integrator
-            integrator.integrator.u = integrator.u
-            integrator.integrator.k = integrator.k
+            # update cached values of u and k
+            if typeof(integrator.u) <: AbstractArray
+                recursivecopy!(integrator.u_cache, integrator.u)
+            else
+                integrator.u_cache = integrator.u
+            end
+            integrator.integrator.u = integrator.u_cache
+
+            recursivecopy!(integrator.integrator.k, integrator.k)
 
             numiters += 1
         end
 
-        # reset values of DDE integrator after last iteration
-        integrator.t = t_cache
-        integrator.tprev = tprev_cache
-        integrator.uprev = uprev_cache
-
-        # update current time of ODE integrator
-        integrator.integrator.t = t_cache
+        # reset time of ODE integrator
+        integrator.integrator.t = integrator.t
     else # no iterations
         perform_step!(integrator, integrator.cache)
     end
 
-    #integrator.u = integrator.integrator.u
-    #integrator.fsallast = integrator.integrator.fsallast
-    #if integrator.opts.adaptive
-    #  integrator.EEst = integrator.integrator.EEst
-    #end
+    # update ODE integrator
+    integrator.integrator.u = integrator.u
```

Comment:
To update the ODE integrator. Otherwise `integrator.integrator.u` still contains `u(integrator.t)`, when peforming a step without iterations, or `u(integrator.t+integrator.dt)` after the second to last iteration, when performing fixed-point iterations. For any interpolations that occur after this step it should rather be the current, updated value `u(integrator.t+integrator.dt)`.